sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java

```diff
@@ -557,6 +533,52 @@ private Assign(WindowingStrategy updatedStrategy) {
     public WindowFn<T, ?> getWindowFn() {
       return updatedStrategy.getWindowFn();
     }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      super.populateDisplayData(builder);
+      populateWindowingStrategyDisplayData(updatedStrategy, builder);
+    }
+  }
+
+  private static void populateWindowingStrategyDisplayData(
+      WindowingStrategy<?, ?> windowingStrategy, DisplayData.Builder builder) {
+
+    builder
```

Comment:
The windowFn and closing behavior are not conditionalized the same way. Cleaner would be to explicitly represent a diff vs a completed strategy.